doc/rlusd-ethereum-design.md

```diff
@@ -0,0 +1,78 @@
+
+## On-Chain roles
+
+Ripple USD supports a number of roles which control how accounts interact with the token on-chain:
+
+- `Minter`: Gives an account the ability to issue stablecoin.
+
+- `Burner`: Gives an account the ability to burn stablecoin.
+
+- `Pauser`: Gives an account the ability to create an individual or global freeze/unfreeze.
+
+- `Clawbacker`: Gives an account the ability to force burn stablecoin from any account.
+
+- `Upgrader`: Gives an account the ability to point the proxy at a new ERC-20, effectively upgrading the contract.
+
+## Issuing and destroying stablecoin
+
+Ripple is the only entity given the ability to _mint_ and _burn_ Ripple USD stablecoin via the `Minter` and `Burner` roles.
+
+Only Ripple's issuer account can call the `mint()` function to issue new stablecoin upon a _distribution request_ from an onboarded customer.
+
+Multiple internal Ripple accounts can call the `burn()` function to ensure an efficient operational process. Users can submit a _redemption request_ to a Ripple owned redemption account. Once the request is received and processed, Ripple will burn the amount of stablecoin in the request.
+
+This iterative process ensures Ripple USD is always fully collateralized.
+
+## Multi-Signing
+
+Security of internal accounts is incredibly important. For this reason, Ripple has chosen to use on-chain multi-signature safeguards for _all_ internal accounts. Since this is supported natively on the XRP Ledger, Ripple chose to expand this functionality to Ethereum by introducing a custom `MultiSign` contract.
+
+The `MultiSign` contract requires the creation of a predetermined list of known signers (accounts). The contract verifies that the transaction bytes provided are signed by the correct accounts, and then forwards the request function to the ERC-20 contract to execute the transaction.
+
+> [!IMPORTANT]
+> This functionality is not required by any community developer, but provides assurance that only Ripple groups will be able to increase and decrease the amount of stablecoin in circulation.
+
+## Events
+
+Events are emitted when the state of the stablecoin contract changes.
+
+- `SignersChanged(address, address[], uint8[], uint256)` : The event is emitted when the `setSigners` method on the contract is called. The values here are the account `address` for which the signers were changed, signer addresses, their weights, and the quorum.
+
+- `Transfer(address, address, uint256)`: Emitted when value is moved from one account to another.
+
+- `Paused(address)`: Emitted when `pause()` is called, triggering a _Global Freeze_ with the address that called the method.
+
+- `Unpaused(address)`: Emitted when `unpause()` is called, triggering a _Global Unfreeze_ with the address that called the method.
+
+- `AccountPaused(address)`: Emitted when `pauseAccount(address)` is called. The `address` represents the account that needs to be _frozen_.
```

Comment:
Since the event is emitted after the change has been successfully applied to the contract, we can say: 
`The address represents the account that was _frozen_.`